### PR TITLE
Consensus: v36 upgrade

### PR DESCRIPTION
**Pull request by jannotti**
Allows upgrade to v36, with boxes.

### CODE REVIEW

File: config/consensus.go [hunk 1]
```diff
@@ -1228,19 +1228,27 @@ func initConsensusProtocols() {
...
+	v36.LogicSigVersion = 8
+	v36.MaxBoxSize = 32768
+	v36.BoxFlatMinBalance = 2500
+	v36.BoxByteMinBalance = 400
+	v36.MaxAppBoxReferences = 8
+	v36.BytesPerBoxReference = 1024
+
+	Consensus[protocol.ConsensusV36] = v36
+
+	v35.ApprovedUpgrades[protocol.ConsensusV36] = 10000
```
Comment (onetechnical) [inline config/consensus.go:1244]: Should be 140000?

File: ledger/testing/consensusRange_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// TestReleasedVersion ensures that the necessary tidying is done when a new
+// protocol release happens.  The new version must be added to
+// consensusByNumber, and a new LogicSigVersion must be added to vFuture.
+func TestReleasedVersion(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	// This confirms that the proto before future has no ApprovedUpgrades.  Once
+	// it does, that new version should be added to consensusByNumber.
+	require.Len(t, config.Consensus[consensusByNumber[len(consensusByNumber)-2]].ApprovedUpgrades, 0)
```
Comment (michaeldiamant) [inline ledger/testing/consensusRange_test.go:37]: @jannotti To help my understanding:  Should `TestReleasedVersion` pass in `master` prior to this PR?

Running locally, I see the highlighted assertion fails.  I'm unfamiliar enough with the details here that it'll take me some time to triage myself.

Comment (michaeldiamant) [reply ledger/testing/consensusRange_test.go:37]: @jannotti Disregard + Resolving - I think I understand now.  It seems the v35 was _not_ properly (?) added to `consensusByNumber`.

Adding v35 to `consensusByNumber` in `master` enables the test to pass.

**Review by algorandskiy — DISMISSED** [review]

**Review by michaeldiamant — APPROVED** [review]
@jannotti Looks accurate to me though I have little experience managing consensus upgrades.  I prefer 1+ other approvals prior to merging.

**Review by gmalouf — APPROVED** [review]

**Pull request merged**